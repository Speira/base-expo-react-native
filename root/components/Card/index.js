import PropTypes from 'prop-types'
import styled from 'styled-components/native'
import { View, Text } from 'react-native'

import BaseCard, { CardHeader, CardBody, CardText, CardFooter } from './style'

/**
 * Card
 * @desc ::: create a layout for the qpp
 *
 */
export default function Card(props) {
  const { title, ...restProps } = props
  return (
    <BaseCard {...restProps}>
      <CardHeader>
        <CardText>{title}</CardText>
      </CardHeader>
      <CardBody />
      <CardFooter />
    </BaseCard>
  )
}
Card.defaultProps = {
  title: '',
}
Card.propTypes = {
  title: PropTypes.string,
}
