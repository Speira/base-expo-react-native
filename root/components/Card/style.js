import styled from 'styled-components/native'
import { ImageBackground, View, Text } from 'react-native'

/**
 * CardHeader
 * @component
 *
 */
export const CardHeader = styled(View)`
  flex: 1;
  justify-content: center;
  padding: 0 8px;
`
/**
 * CardBody
 * @component
 *
 */
export const CardBody = styled(View)`
  flex: 4;
  padding: 4px;
  justify-content: center;
`

/**
 * CardBodyImage
 * @component
 *
 */
export const CardBodyImage = styled(ImageBackground)`
  flex: 1;
  justify-content: center;
`
/**
 * CardFooter
 * @component
 *
 */
export const CardFooter = styled(View)`
  flex: 1;
  padding: 4px;
  justify-content: center;
`
/**
 * CardText
 *
 */
export const CardText = styled(Text)`
  color: ${({ theme }) => theme.COLORS.STATIC.DARK};
  flex-wrap: wrap;
`
/**
 * BaseCard
 * @component
 *
 */
const BaseCard = styled(View)`
  height: 300px;
  width: 300px;
  background-color: ${({ theme }) => theme.COLORS.STATIC.QUATERNARY};
  border-radius: 4px;
  padding: 4px;
  text-align: center;
  shadow-color: ${({ theme }) => theme.COLORS.STATIC.DARK};
  shadow-offset: {width: 1, height: 2};
  shadow-opacity: 0.6;
  shadow-radius: 4px;
  margin: 8px;
`
export default BaseCard
