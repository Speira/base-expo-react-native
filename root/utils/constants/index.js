// utils/constants
// description: We try to set all constants into a big one to reduce Javascript
//              constants performance issue.

export default {
  API: {
    FETCH_DATA_URL:
      'https://static.leboncoin.fr/img/shared/technical-test.json',
    PROXY: {
      NO_CORS_SERVER: 'https://cors-anywhere.herokuapp.com/',
    },
    TIMEOUT: 20000,
  },
  LABELS: {
    CLEAR_DATABASE: 'Clear the database',
    START_DATABASE: 'Create the database',
    IMVALID_DATA_FORMAT: ' Invalid data format',
    LIST: 'Item List',
    NO_CARD: 'No card to persist',
    NO_DATABASE: 'The database is not set',
    NO_RESPONSE_DATA: 'There is no data in the query response',
    REFETCH_DATA: 'Refetch Data',
    SEARCH_DATA_FAILED:
      'No data for SearchDataAPI: Please unlock the proxy below and then refetch data',
    WELCOME: 'Welcome to the ListItem app',
  },
  STATES: {
    ERROR: 'error',
    LOADING: 'loading',
  },
  STATUS: {
    DANGER: 'DANGER',
    INFO: 'INFO',
    SUCCESS: 'SUCCESS',
    WARNING: 'WARNING',
  },
  STORAGE_ITEMS: {
    DB: 'storage_list_item_app',
    CARD: 'card_items.db',
    THEME: 'app_theme_storage',
  },
  THEMES: {
    DEFAULT: {
      COLORS: {
        STATIC: {
          PRIMARY: '#EBD8C3',
          SECONDARY: '#F7E9D7',
          TERTIARY: '#FFF6EA',
          QUATERNARY: '#FFEEEE',
          LIGHT: '#f8f8fa',
          DARK: '#4d3e30',
        },
        DYNAMIC: {
          INFO: '#035397',
          SUCCESS: '#1C7947',
          WARNING: '#C36A2D',
          DANGER: '#A9333A',
        },
      },
    },
  },
}
