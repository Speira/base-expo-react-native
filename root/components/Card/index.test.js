import React from 'react'
import { shallow } from 'enzyme'
import constants from '~utils/constants'
import Card from './index'

const { THEMES } = constants

describe('render Card', () => {
  const wrapper = shallow(<Card title="test" theme={THEMES.DEFAULT} />)
  it('Card should be defined', () => {
    expect(wrapper).toBeDefined()
  })
  it('Card should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
