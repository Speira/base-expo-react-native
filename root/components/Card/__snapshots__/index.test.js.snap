// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Card Card should match snapshot 1`] = `
<style__BaseCard
  theme={
    Object {
      "COLORS": Object {
        "DYNAMIC": Object {
          "DANGER": "#A9333A",
          "INFO": "#035397",
          "SUCCESS": "#1C7947",
          "WARNING": "#C36A2D",
        },
        "STATIC": Object {
          "DARK": "#4d3e30",
          "LIGHT": "#f8f8fa",
          "PRIMARY": "#EBD8C3",
          "QUATERNARY": "#FFEEEE",
          "SECONDARY": "#F7E9D7",
          "TERTIARY": "#FFF6EA",
        },
      },
    }
  }
>
  <style__CardHeader>
    <style__CardText>
      test
    </style__CardText>
  </style__CardHeader>
  <style__CardBody>
    <style__CardBodyImage
      resizeMode="cover"
      source={
        Object {
          "uri": "",
        }
      }
    />
  </style__CardBody>
  <style__CardFooter />
</style__BaseCard>
`;
